### PR TITLE
Adding SYCL 1.2.1 Compatibility.

### PR DESCRIPTION
**Pull request by mehdi-goli**

### CODE REVIEW

File: include/views/view_sycl.hpp [hunk 1]
```diff
@@ -292,6 +292,8 @@ struct vector_view<ScalarT, bufferT<ScalarT>> {
   }
 };
 
+
```
Comment (lukeiwanski) [inline include/views/view_sycl.hpp]: do we need these?

File: tests/paper_blas1_test.cpp [hunk 2]
```diff
@@ -494,6 +494,28 @@ int main(int argc, char *argv[]) {
...
+      buffer<double, 1> bR3(range<1>{vR3.size()});
+      buffer<double, 1> bS3(range<1>{vS3.size()});
+      buffer<double, 1> bX4(range<1>{vX4.size()});
+      buffer<double, 1> bY4(range<1>{vY4.size()});
+      buffer<double, 1> bZ4(range<1>{vZ4.size()});
+      buffer<double, 1> bR4(range<1>{vR4.size()});
+      buffer<double, 1> bS4(range<1>{vS4.size()});
+
+      /*
       buffer<double, 1> bX1(vX1.data(), range<1>{vX1.size()});
```
Comment (lukeiwanski) [inline tests/paper_blas1_test.cpp]: commented out code.

Comment (mehdi-goli) [reply tests/paper_blas1_test.cpp]: This is deliberately commented  out. In order to test the issue we are seeing in Computecpp.

File: tests/paper_blas1_test.cpp [hunk 3]
```diff
@@ -560,25 +583,25 @@ int main(int argc, char *argv[]) {
...
+
+        h.copy((vZ1.data()), accZ1);
+        h.copy((vZ2.data()), accZ2);
+        h.copy((vZ3.data()), accZ3);
+        h.copy((vZ4.data()), accZ4);
+
+        h.copy((vR1.data()), accR1);
+        h.copy((vR2.data()), accR2);
+        h.copy((vR3.data()), accR3);
+        h.copy((vR4.data()), accR4);
```
Comment (Ruyk) [inline tests/paper_blas1_test.cpp]: why the extra parenthesis?

File: tests/paper_blas1_test.cpp [hunk 4]
```diff
@@ -560,25 +583,25 @@ int main(int argc, char *argv[]) {
...
-        h.update_to_device(accR2);
-        h.update_to_device(accR3);
-        h.update_to_device(accR4);
+        h.copy((vX1.data()), accX1);
+        h.copy((vX2.data()), accX2);
+        h.copy((vX3.data()), accX3);
+        h.copy((vX4.data()), accX4);
+
+        h.copy((vY1.data()), accY1);
+        h.copy((vY2.data()), accY2);
```
Comment (Ruyk) [inline tests/paper_blas1_test.cpp]: According to SYCL 1.2.1, each copy should go on a separate command group

**Comment by Ruyk** [comment]
The travis needs updating, simply use the configuration from the Khronos SYCL Parallel STL project

**Pull request merged**